frontend/src/lib/Utils.tsx

```diff
@@ -358,3 +359,24 @@ export function buildQuery(queriesMap: { [key: string]: string | number | undefi
   }
   return `?${queryContent}`;
 }
+
+export async function decodeCompressedNodes(compressedNodes: string): Promise<object> {
+  return new Promise<object>(resolve => {
+    const compressedBuffer = Buffer.from(compressedNodes, 'base64');
+    zlib.gunzip(compressedBuffer, { windowBits: 15 }, (error, result: Buffer) => {
+      if (error) {
+        logger.error('failed to gunzip data ', error);
+        resolve(undefined);
```

Comment:
nit: a more common practice is to reject when there's an error, but it's up to you whether you want to change

Reply:
You're right, I will reject here and catch error in RunDetails.